src/main.js

```diff
@@ -98,13 +98,51 @@ class Entry {
    *@param {string|null} defaultValue given default value e.g. data
    *@param {string} comment given comment using the # symbol
    */
+
   constructor(name, type, choices, defaultValue, comment) {
     this.name = name;
     this.type = type;
     this.choices = choices;
     this.defaultValue = defaultValue;
     this.comment = comment;
   }
+
+  /** @method html
+   * @summary Returns the rendered HTML for the given entry
+   * @returns {promise} that resolves to string
+   */
+  html() {
+    return new Promise(function(resolve, reject) {
+      if (this) {
+        resolve(outputHTML(this));
+      } else {
+        reject("Error:Wrong Syntax");
+      }
+    });
+  }
+}
+
+/**
+ * Class representing an array of entry objects
+ * @class
+ */
+class EntryList {
+  /**
+   * @param {Array} entries includes {@link Entry} objects
+   */
+  constructor(entries) {
+    this.entries = entries;
+  }
+
+  /** @method html
+   * @summary Returns the rendered HTML for the given array of entries
+   * @returns {promise} that resolves to string
+   */
+  html() {
+    return Promise.all(this.entries.map(entry => entry.html())).then(values =>
```

Comment:
Here is the trick. What you need to do is return a promise, that will execute `Promise.all` itself 🤯.

It should look like this

```js
return new Promise((resolve, reject) => {
  Promise.all(this.entries.map(...))
    .then(values => resolve(values.join("\n")))
    .catch(error => reject(error);
});
```

(or something like that)